src/vm/clarity.rs

```diff
@@ -152,16 +248,20 @@ impl <'a> ClarityBlockConnection <'a> {
     /// Save a contract analysis output to the AnalysisDatabase
     /// An error here would indicate that something has gone terribly wrong in the processing of a contract insert.
     ///   the caller should likely abort the whole block or panic
-    pub fn save_analysis(&mut self, identifier: &QualifiedContractIdentifier, contract_analysis: &ContractAnalysis) -> Result<(), CheckError> {
+    pub fn save_analysis(&mut self, identifier: &QualifiedContractIdentifier, contract_analysis: &ContractAnalysis) -> Result<(), Error> {
```

Comment:
I don't think we should reduce the specificity of the return type here -- this function _can only ever_ throw a CheckError, there's no reason for this function to return a different type of error when any caller can either use `Error::from` or `.into` or `?` to handle the conversion, whereas a caller that is expecting to handle a CheckError would need to do a spurious match with unreachable paths.

Reply:
Sure, I can fix that.  IIRC I made that change to make it easier to write the transaction-processing code -- this change is not functionally required.